File: Programmers/lv1/lv1_하샤드 수.java
https://programmers.co.kr/learn/courses/30/lessons/12947

class Solution {
    public boolean solution(int x) {
        boolean answer = true;
        
        int sum = 0;
        int num = x;
        
        while(num != 0){
            sum += num%10;
            num /= 10;
        }

        if(x%sum != 0){
            answer = false;
        }
        
        return answer;
    }
}

File: Programmers/C#/lv1/lv1_하샤드 수.cs
https://programmers.co.kr/learn/courses/30/lessons/12947?language=csharp

public class Solution
{
    public bool solution(int x)
    {
        bool answer = true;

        string temp = x.ToString();
        int sum = 0;

        foreach (char c in temp)
        {
            sum += int.Parse(c.ToString());
        }

        answer = (x % sum == 0) ? true : false;

        return answer;
    }
}